frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { AsyncLoader } from '../../utils/async';
+
+type CloseModal = () => void;
+
+export type LaunchModal = (
+  component: () => Promise<React.FC<{ closeModal: CloseModal }>>,
+  extraProps: any,
+) => void;
+
+type ModalProps = {
+  launchModal: LaunchModal;
+  closeModal: CloseModal;
+};
+
+const defaultModalContext: ModalProps = {
+  launchModal: null,
+  closeModal: null,
+};
+
+export const ModalContext = React.createContext(defaultModalContext);
+
+export const ModalProvider = ({ children }) => {
+  const [isOpen, setOpen] = React.useState(false);
+  const component = React.useRef(null);
+  const [componentProps, setComponentProps] = React.useState({});
+
+  const launchModal = React.useCallback(
+    (comp, compProps) => {
+      component.current = comp;
+      setComponentProps(compProps);
+      setOpen(true);
+    },
+    [setOpen, setComponentProps],
+  );
+  const closeModal = React.useCallback(() => setOpen(false), [setOpen]);
+
+  return (
+    <ModalContext.Provider value={{ launchModal, closeModal }}>
+      {isOpen && (
+        <AsyncLoader loader={component.current} {...componentProps} closeModal={closeModal} />
+      )}
+      {children}
```

Comment:
The modal should come after the children.

Reply:
ack